src/VisualStudio/Core/Def/Implementation/Venus/ContainedDocument.cs

```diff
@@ -1031,7 +1054,7 @@ private RazorCodeBlockType GetRazorCodeBlockType(int position)
         {
             var subjectBuffer = (IProjectionBuffer)this.GetOpenTextBuffer();
```

Comment:
Maybe an assert that `_hostType == HostType.Razor` so that the case below is safe?